python/ray/workflow/step_executor.py

```diff
@@ -194,17 +221,62 @@ async def _write_step_inputs(wf_storage: workflow_storage.WorkflowStorage,
     await asyncio.gather(*save_tasks)
 
 
+def _validate_checkpoint_options(ret: Union["Workflow", Any]):
+    """Validate checkpoint options. This implementation maybe not most
```

Comment:
Add some comments about what you are checking here. For example, which condition is not allowed to help future maintainers.